From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ossi <oser0002@gmail.com>
Date: Fri, 12 Jun 2020 01:38:06 +0300
Subject: [PATCH] Fix CraftScheduler#runTaskTimerAsynchronously(Plugin,
 Consumer<BukkitTask>, long, long) scheduling a non-repeating task instead of
 a repeating one.


diff --git a/src/main/java/org/bukkit/craftbukkit/scheduler/CraftScheduler.java b/src/main/java/org/bukkit/craftbukkit/scheduler/CraftScheduler.java
index ea7ebbc2674df727cf44856f172731ee083b8800..a423970cf7c927ea8a1bf842aaa236d3cf2d54c2 100644
--- a/src/main/java/org/bukkit/craftbukkit/scheduler/CraftScheduler.java
+++ b/src/main/java/org/bukkit/craftbukkit/scheduler/CraftScheduler.java
@@ -199,7 +199,7 @@ public class CraftScheduler implements BukkitScheduler {
 
     @Override
     public void runTaskTimerAsynchronously(Plugin plugin, Consumer<BukkitTask> task, long delay, long period) throws IllegalArgumentException {
-        this.runTaskTimerAsynchronously(plugin, (Object) task, delay, CraftTask.NO_REPEATING);
+        this.runTaskTimerAsynchronously(plugin, (Object) task, delay, period);
     }
 
     @Override
